Framework/src/Bookkeeping.cxx

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   Bookkeeping.cxx
+/// \author Barthelemy von Haller
+///
+
+#include "QualityControl/Bookkeeping.h"
+#include "QualityControl/QcInfoLogger.h"
+#include "BookkeepingApi/BkpProtoClientFactory.h"
+
+using namespace o2::bkp::api::proto;
+
+namespace o2::quality_control::core
+{
+
+void Bookkeeping::init(std::string url)
+{
+  if (url.empty()) {
+    ILOG(Warning, Support) << "No URL provided for Bookkeeping. Nothing will be stored nor retrieved." << ENDM;
+    return;
+  }
+  try {
+    mClient = BkpProtoClientFactory::create(url);
```

Comment:
Can `create` return a nullptr instead of throwing an exception? Are we guaranteed that this does not happen in the future? I am afraid we might end up with a code path which has `mInitialized == true`, but `mClient == nullptr`.